Source/WebCore/rendering/style/StyleImageSet.cpp

```diff
@@ -90,14 +90,16 @@ ImageWithScale StyleImageSet::bestImageForScaleFactor()
         const auto& image = m_images[index];
         if (!image.mimeType.isNull() && !MIMETypeRegistry::isSupportedImageMIMEType(image.mimeType))
             continue;
-
+        if (!result.image->isInvalidImage() && result.scaleFactor == image.scaleFactor)
+            continue;
         if (image.scaleFactor >= m_deviceScaleFactor)
             return image;
+
         result = image;
     }
 
     ASSERT(result.scaleFactor >= 0);
-    if (!result.image || !result.scaleFactor)
+    if (result.image->isInvalidImage() || !result.scaleFactor)
```

Comment:
I feel like we could just remove this invalid image check, since it's mostly no-op in that case, unless you see a compelling reason to keep it.

Reply:
This check catches the case where only unsupported types were provided to the set and we never assigned to `result` in the loop:

`image-set('foo.invalid' type('image/invalid'));`

Reply:
Another option might be to change the default scaleFactor to 0 in `ImageWithScale` but that default predates the work I've been doing on image-set and I have no idea what the fallout might be.